lib/resValidator.js

```diff
@@ -111,7 +111,16 @@ function validateResAgainstApiSpec(res, expectedResponse, openApiSpec) {
 
 
 function getOAPath(req, openApiSpec) {
-  const pathname = url.parse(req.path).pathname; // excludes the query (because: path = pathname + query)
+  let pathname = url.parse(req.path).pathname; // excludes the query (because: path = pathname + query)
+
+  const serverUrl = getServerUrlUsed(pathname, openApiSpec.servers);
+
+  // if servers have been defined but none of the servers match the current path
+  if (openApiSpec.servers && !serverUrl) {
+    throw new Error(`No server matching '${pathname}' path defined in OpenAPI spec`);
+  }
+  pathname = removeServerURLFromPath(pathname, serverUrl);
```

Comment:
nit: maybe call this `const pathnameWithoutServerUrl` so the above `pathname` can be a `const` too?

Reply:
Yep that's clearer